method.rb

```diff
@@ -0,0 +1,218 @@
+        self.my_each do |x|
+          return false unless pattern.match(x.to_s)
+        end
+      else
+        self.my_each do |x|
+          return false unless x == pattern
+        end
+      end
+    else
+      self.my_each do |x|
+        return false unless x
+      end
+    end
+    true
+  end
+
+  def my_any?(patter = nil)
+    if block_given?
+      self.my_each do |x|
+        return true if yield(x)
+      end
+    elsif !patter.nil?
+      if patter.is_a?(Class)
+        self.my_each do |x|
+          return true if x.is_a?(patter)
+        end
+      elsif patter.is_a?(Regexp)
+        self.my_each do |x|
+          return true if patter.match(x.to_s)
+        end
+      else
+        self.my_each do |x|
+          return true if x == patter
+        end
+      end
+    else
+      self.my_each do |x|
+        return true if x
+      end
+    end
+    false
+  end
+
+  def my_none?(pat = nil)
+    if block_given?
+      self.my_each do |x|
+        return false if yield(x)
+      end
+    elsif !pat.nil?
+      if pat.is_a?(Class)
+        self.my_each do |x|
+          return false if x.is_a?(pat)
+        end
+      elsif pat.is_a?(Regexp)
+        self.my_each do |x|
+          return false if pat.match(x.to_s)
+        end
+      else
+        self.my_each do |x|
+          return false if x == pat
+        end
+      end
+    else
+      self.my_each do |x|
+        return false if x
+      end
+    end
+    true
+  end
+
+  def my_count(element = nil)
+    total = 0
+    if block_given?
+      self.my_each do |x|
+        total += 1 if yield(x)
+      end
+    elsif !element.nil?
+      self.my_each do |x|
+        total += 1 if x == element
+      end
+    else
+      total = self.size
+    end
+    total
+  end
+
+  def my_map
+    if block_given?
+      arr = []
+      if proc.nil?
+        self.my_each { |x| arr << yield(x) }
+      else
+        self.my_each { |x| arr << proc.call(x) }
+      end
+      arr
+    else
+      to_enum(:my_map)
+    end
+  end
+
+  # def my_inject(init = nil, sym = nil)
+  #   i = 1
+  #   if init.nil?
+  #     init = self[0]
+  #     i = 1
+  #   end
+  #   if init && sym.nil?
+  #     if init.class == Symbol
+  #       symbol = init
+  #       init = self[0]
+  #       i = 1
+  #       while i < self.size
+  #         init = self[i].method(symbol).call(init)
+  #         i += 1
+  #       end
+  #     elsif init.class == Integer
+  #       while i < self.size
+  #         init = yield(init, self[i])
+  #         i += 1
+  #       end
+  #     end
+  #   elsif init && sym
+  #     i = 0
+  #     while i < self.size
+  #       init = self[i].method(sym).call(init)
+  #       i += 1
+  #     end
+  #   elsif !init && !sym
+  #     raise ArgumentError, 'Incorrect arguments provided'
+  #   else
+  #     init = self[0]
+  #     i = 1
+  #     while i < self.size
+  #       init = yield(init, self[i])
+  #       i += 1
+  #     end
+  #   end
+  #   init
+  # end
```

Comment:
**Always remember to delete commented code or debugging information before pushing your code to the repo.** 
If you need to review a previous version of your code, use git history.